app/src/main/java/com/example/toyproject/viewmodel/MySkinViewModel.kt

```diff
@@ -0,0 +1,76 @@
+package com.example.toyproject.viewmodel
+
+import androidx.hilt.lifecycle.ViewModelInject
+import androidx.lifecycle.*
+import com.example.toyproject.base.BaseProvider
+import com.example.toyproject.data.db.entities.Skins
+import com.example.toyproject.domain.repository.SkinRepository
+import com.example.toyproject.util.Resource
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.FlowPreview
+import kotlinx.coroutines.channels.BroadcastChannel
+import kotlinx.coroutines.channels.Channel
+import kotlinx.coroutines.flow.*
+import kotlinx.coroutines.launch
+
+class MySkinViewModel @ViewModelInject constructor(
+    private val skinRepository: SkinRepository,
+    private val dispatchers: BaseProvider,
+    val skinTitle: String
+) : ViewModel() {
+    private val _skinLiveData = MutableLiveData<Resource<List<Skins>>>()
+    val skinLiveData: LiveData<Resource<List<Skins>>>
+        get() = _skinLiveData
+
+    fun loadAllMySkin() = viewModelScope.launch {
+        _skinLiveData.postValue(Resource.loading())
+        skinRepository.loadAllSkins()
+            .flowOn(dispatchers.default)
+            .catch { e -> _skinLiveData.postValue(Resource.Error(e)) }
+            .collect {
+                _skinLiveData.postValue(Resource.success(it))
+            }
+    }
+
+    fun fetchBySkinTitle(skinTitle: String) = viewModelScope.launch {
+        _skinLiveData.postValue(Resource.loading())
+        skinRepository.loadBySkinTitle(skinTitle)
+            .flowOn(dispatchers.default)
+            .catch { e -> _skinLiveData.postValue(Resource.error(e)) }
+            .collect { _skinLiveData.postValue(Resource.Success(it)) }
+    }
+
+    fun deleteAllSkins() = viewModelScope.launch(dispatchers.default) {
+        skinRepository.deleteAllSkins()
+    }
```

Comment:
```suggestion
    fun deleteAllSkins() = viewModelScope.launch {
        skinRepository.deleteAllSkins()
    }
```
요렇게 쓰면 안되는 건가용 ? 궁금해서 여쭈어봐요 ~

Reply:
delete이니 Room에 접근 하는 코드인데 Dispathcers.IO에서 접근하는게 나아 보여요

Reply:
코루틴 스코프를 지정해주는 코드인데, 디스패처를 지정해주지 않아도 에러가 발생하진 않지만 저 코드가 동작할 때 Room에 접근하는 코드이기 때문에 디스패처를 지정해주는 게 좋다고 생각했어용